patches/0049-non-idempotent-sns-topic-creation.patch

```diff
@@ -0,0 +1,52 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: corymhall <43035978+corymhall@users.noreply.github.com>
+Date: Wed, 10 Apr 2024 10:02:02 -0400
+Subject: [PATCH 49/49] non-idempotent sns topic creation
+
+
+diff --git a/internal/service/sns/topic.go b/internal/service/sns/topic.go
+index 98ebd87d5d..5f3a61390b 100644
+--- a/internal/service/sns/topic.go
++++ b/internal/service/sns/topic.go
+@@ -9,6 +9,7 @@ import (
+ 	"fmt"
+ 	"log"
+ 	"regexp"
++	"sync"
+ 	"time"
+ 
+ 	"github.com/YakDriver/regexache"
+@@ -259,6 +260,12 @@ func resourceTopic() *schema.Resource {
+ 	}
+ }
+ 
++func constructTopicArn(client *sns.Client, account, region, snsTopicName string) string {
++	return fmt.Sprintf("arn:aws:sns:%s:%s:%s", region, account, snsTopicName)
++}
++
++var snsGlobalMutex sync.Mutex
```

Comment:
I know I've approved this before but is this is going to be pretty nasty for programs that create 1000 topics this mutex will create a global chokepoint if there wasn't one already, so they will be literally created one at a time?

Do we just need locks for a particular topic? in that case maybe we could have a sync.Map of murexes indexed by topic ARN here so as not to conflict globally? 

CC @VenelinMartinov

Reply:
@t0yv0 I've updated this so that the lock is based on the topic arn. I think we only care to prevent race conditions if the topics have the same name.